### DISCUSSION TITLE
JavaScript heap out of memory error when Migration to 3.3.5

### DISCUSSION THREAD

**Discussion by pankajthamkeappdirect** [Netflix/conductor#2719] [Q&A]
Hi, I've been working on migrating Conductor 1.8.x -> Conductor 3.3.5.  Conductor-UI is failing with JavaScript heap out of memory. Please find the logs.

```
yarn run v1.22.15
warning package.json: No license field
warning Skipping preferred cache folder "/home/node/.cache/yarn" because it is not writable.
warning Selected the next writable cache folder in the list, will be "/tmp/.yarn-cache-1000".
$ react-scripts start
warning Cannot find a suitable global folder. Tried these: "/usr/local, /home/node/.yarn"
[HPM] Proxy created: /  -> http://localhost:8080
[34mℹ[39m [90m｢wds｣[39m: Project is running at http://10.26.103.180/
[34mℹ[39m [90m｢wds｣[39m: webpack output is served from 
[34mℹ[39m [90m｢wds｣[39m: Content not from webpack is served from /app/ui/public
[34mℹ[39m [90m｢wds｣[39m: 404s will fallback to /
Starting the development server...


<--- Last few GCs --->

[34:0x7f99bb8eb2c0]   125312 ms: Scavenge (reduce) 253.1 (256.5) -> 252.7 (257.8) MB, 1.5 / 0.0 ms  (average mu = 0.192, current mu = 0.002) allocation failure 
[34:0x7f99bb8eb2c0]   127098 ms: Mark-sweep (reduce) 253.5 (256.8) -> 253.1 (258.5) MB, 1700.9 / 0.1 ms  (+ 79.2 ms in 13 steps since start of marking, biggest step 73.0 ms, walltime since start of marking 1785 ms) (average mu = 0.190, current mu = 0.188)

<--- JS stacktrace --->

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by pankajthamkeappdirect** [comment]
Someone please help me with this.

**Comment by jxu-nflx** [comment]
@peterlau Do you have any insight about this?

**Reply by pankajthamkeappdirect** [reply]
@peterlau  Our client is impacted because of this, It will be very helpful if you provide some suggestions. Below is my docker file with logs.

`FROM openjdk:11-jdk

RUN mkdir -p /build_output
WORKDIR /build_output

RUN git clone https://github.com/Netflix/conductor.git \
  && cd conductor \
  && git checkout -b 3.3.5 v3.3.5 \
  && ./gradlew build -x test --stacktrace \
  && cp ./server/build/libs/conductor-server-*-SNAPSHOT.jar /build_output/

ENTRYPOINT ./gradlew build -x test --stacktrace

FROM openjdk:11-jre
MAINTAINER PSDS team <psds@appdirect.com>

FROM alpine:3.4
RUN apk add --no-cache bash

RUN mkdir -p /app/config /app/logs /app/libs

COPY --from=0 /build_output/*.jar /app/libs
COPY ./bin /app
COPY ./config /app/config

RUN chmod +x /app/startup.sh

EXPOSE 8080

CMD ["/app/startup.sh"]
ENTRYPOINT ["/bin/bash"]`

Logs:
`waiting for elasticsearch ...
/app/startup.sh: line 48: java: not found
Starting Conductor Server
Detected overridden value for: CONDUCTOR_DB=mysql
Detected overridden value for: JDBC_URL=jdbc:mysql://od26-db.ctl6ajek5jah.us-east-1.rds.amazonaws.com:3306/kb_p39pwymnm_cdt?useSSL=false
Detected overridden value for: JDBC_USERNAME=kb_p39pwymnm_cdt
Detected overridden value for: JDBC_PASSWORD=**hidden**
Detected overridden value for: ELASTICSEARCH_URL=localhost:9300
Detected overridden value for: ELASTICSEARCH_INDEX=conductor
Detected overridden value for: LOAD_SAMPLE=false`

**Reply by pankajthamkeappdirect** [reply]
Please someone help here. If possible can we have quick google meet call to fix this.
